### ISSUE TITLE
Support reproducible flow-build-info.json and stats.json

### ISSUE THREAD

**Issue by lightoze** [vaadin/flow#7556]
Currently (v14) generated `flow-build-info.json` and `stats.json` entries are randomly reordered during build. In containerized environment it means that every build produces an unique image which creates problems both in development and CD workflows.

This should be also taken into account in relation to #6398

**Comment by denis-anisimov** [comment]
As I understand you would like to have the same content of the `flow-build-info.json`  and `stats.json` files, right ?

**Comment by lightoze** [comment]
@denis-anisimov Right. It appears to me that the only difference I get now is content ordering.

**Comment by denis-anisimov** [comment]
OK, thanks for clarification.

**Comment by lightoze** [comment]
I've got back to this issue, tested with Vaadin 21.0.6, `flow-build-info.json` still seems to be randomly reordered after each build. When building to containers it forces a new set of layers to be created on each build, even when particular application in multi-module repository is unchanged.

For testing purposes I've created a test project which allows quickly deploying a test app into a Kubernetes cluster using Skaffold: https://github.com/lightoze/vaadin-build-demo. After byte to byte comparison of images produces by a clean build it is seen that `flow-build-info.json` is the only file that differs between the builds.
